managed/src/main/java/com/yugabyte/yw/controllers/EncryptionAtRestController.java

```diff
@@ -44,13 +49,44 @@
 
 public class EncryptionAtRestController extends AuthenticatedController {
     public static final Logger LOG = LoggerFactory.getLogger(EncryptionAtRestController.class);
+    static final String SMARTKEY_KMS_PROVIDER = "SMARTKEY";
+    static final String AWS_KMS_PROVIDER = "aws";
 
     @Inject
     EncryptionAtRestManager keyManager;
 
     @Inject
     Commissioner commissioner;
 
+    @Inject
+    CloudAPI.Factory cloudAPIFactory;
+
+    private void validateKMSProviderConfigFormData(ObjectNode formData, String keyProvider) {
+      if(keyProvider.toLowerCase().equals(AWS_KMS_PROVIDER)
+          && (formData.get("AWS_ACCESS_KEY_ID")!=null
+          || formData.get("AWS_SECRET_ACCESS_KEY")!=null)) {
+        CloudAPI cloudAPI = cloudAPIFactory.get(AWS_KMS_PROVIDER);
+        Map<String, String> config = new HashMap<String,String>();
+        config.put("AWS_ACCESS_KEY_ID", formData.get("AWS_ACCESS_KEY_ID").textValue());
+        config.put("AWS_SECRET_ACCESS_KEY", formData.get("AWS_SECRET_ACCESS_KEY")
+            .textValue());
+        if (cloudAPI != null && !cloudAPI.isValidCreds(config, formData.get("AWS_REGION")
```

Comment:
Where is this isValidCreds method defined? I can't seem to find it.

Reply:
Also, we should do some pre-flight checks on if the customer has chosen to use IAM instance profile credentials for their AWS KMS config instead of explicitly providing credentials. See AwsEARServiceUtil.java for how we fallback to using those credentials if a user has not explicitly provided any.

Reply:
**isValidCreds** is present in CloudAPI.java and its implementation is given in AWSCloudImpl.java.

Reply:
This method got recently merged as part of separate PR https://github.com/yugabyte/yugabyte-db/pull/8393